packages/network/src/v2/schemas/decodeKeyTuple.ts

```diff
@@ -1,11 +1,16 @@
-import { SchemaTypeToAbiType } from "@latticexyz/schema-type";
+import { StaticSchemaType, getStaticByteLength } from "@latticexyz/schema-type";
+
 import { Schema } from "../common";
-import { decodeAbiParameters } from "viem";
+import { decodeStaticField } from "./decodeStaticField";
+import { hexToArray } from "@latticexyz/utils";
 
 export function decodeKeyTuple(keySchema: Schema, keyTuple: unknown[]) {
-  const abiTypes = keySchema.staticFields.map((schemaType) => SchemaTypeToAbiType[schemaType]);
-  const decodedKeys = keyTuple.map(
-    (key, index) => decodeAbiParameters([{ type: abiTypes[index] }], key as `0x${string}`)[0]
-  );
+  const decodedKeys = keyTuple.map((key, index) => {
+    const schemaType = keySchema.staticFields[index] as StaticSchemaType;
+    const byteLength = getStaticByteLength(schemaType);
+    // Key data is padded, so need offset
+    const offset = 32 - byteLength;
```

Comment:
this is a bit dangerous because a bytes8 value will be padded differently than an int32 value

Reply:
@holic okay just made an extremely small brain fix to that, lmk what you think

Reply:
@holic just a polite lil bump on this (locally linking for now but would prefer merged)

Reply:
I am still worried about the approach here using offsets to account for padding, since the direction of the padding will be different depending on the data type.

I've taken a light week this week after the hackathon but can dig into this next week.

Reply:
Okay, thanks!